core/src/main/java/nu/yona/server/device/service/DeviceService.java

```diff
@@ -105,6 +105,7 @@ public UserDeviceDto addDeviceToUser(User userEntity, UserDeviceDto deviceDto)
 		UserDevice deviceEntity = userDeviceRepository.save(UserDevice.createInstance(userEntity, deviceDto.getName(),
 				deviceAnonymized.getId(), userService.generatePassword()));
 		userEntity.addDevice(deviceEntity);
+		userAnonymizedService.updateUserAnonymized(userEntity.getAnonymized().getId());
```

Comment:
Does this line fix another bug? If so, should we add another test?

Reply:
Yes, the test initially failed to post an activity because the device anonymized didn't get synchronized to the analysis engine service. An extra test isn't necessary, as that's covered with this test.